src/hash_table.py

```diff
@@ -0,0 +1,80 @@
+# -*- coding: utf-8 -*-
+"""
+    This module will implement a hash_table on a string input
+
+    This implementation of the hash table was derived from the examples found
+    below:
+
+    http://www.laurentluce.com/posts/python-dictionary-implementation/
+    http://www.partow.net/programming/hashfunctions/
+
+    Help with correctly selecting the hash function and using bytearray
+    came from:
+
+    http://www.github.com/crashtack
+"""
+
+
+def bp_hash(key, size):
+    hash_value = 0
+    for i in range(len(key)):
+        hash_value = (hash_value << 7 ^ ord(key[i])) % size
+    return hash_value
+
+
+def simple_hash(key, size):
+    key = bytearray(key.encode())
```

Comment:
This encode won't work with strings containing non-ascii characters. Make sure to specify `key.encode("utf-8")` since the default encoding is ascii.